source/rrRoadRunnerOptions.cpp

```diff
@@ -340,19 +334,94 @@ namespace rr {
         if (Config::getBool(Config::LOADSBMLOPTIONS_OPTIMIZE_INSTRUCTION_SIMPLIFIER))
             modelGeneratorOpt |= LoadSBMLOptions::OPTIMIZE_INSTRUCTION_SIMPLIFIER;
 
+        if (Config::getBool(Config::LLVM_SYMBOL_CACHE))
+            modelGeneratorOpt |= LoadSBMLOptions::LLVM_SYMBOL_CACHE;
+
+        // todo delete this options and use the below system instead
         if (Config::getBool(Config::LOADSBMLOPTIONS_USE_MCJIT))
             modelGeneratorOpt |= LoadSBMLOptions::USE_MCJIT;
 
-        if (Config::getBool(Config::LLVM_SYMBOL_CACHE))
-            modelGeneratorOpt |= LoadSBMLOptions::LLVM_SYMBOL_CACHE;
+        // use the values in Config to update values in Options
+        Config::LLVM_BACKEND_VALUES whichBackend =
+                (Config::LLVM_BACKEND_VALUES) Config::getValue(Config::LLVM_BACKEND).getAs<int>();
+
+        // updates modelGeneratorOpt inside setLLVMBackend...
+        switch (whichBackend) {
+            case Config::MCJIT:
+                setLLVMBackend(LoadSBMLOptions::MCJIT);
+                break;
+            case Config::LLJIT:
+                setLLVMBackend(LoadSBMLOptions::LLJIT);
+                break;
+            default:
+                std::string err = "Compiler option is invalid. Defaulting back to MCJit";
+                rrLogWarn << err;
```

Comment:
I think I would make this an error, and might even throw an exception.  If someone said to do something and you don't, chances are they need to know.

Reply:
Yep makes sense.